lib/json/json.go

```diff
@@ -54,15 +54,19 @@ import (
 // (e.g. it implements both Iterable and HasFields), the first case takes precedence.
 // Encoding any other value yields an error.
 //
-// def decode(x):
+// def decode(x[, default]):
 //
-// The decode function accepts one positional parameter, a JSON string.
+// The decode function has one required positional parameter, a JSON string.
 // It returns the Starlark value that the string denotes.
 // - Numbers are parsed as int or float, depending on whether they
 //   contain a decimal point.
 // - JSON objects are parsed as new unfrozen Starlark dicts.
 // - JSON arrays are parsed as new unfrozen Starlark lists.
-// Decoding fails if x is not a valid JSON string.
+// If `x` is not a valid JSON string and the optional, positional-only `default`
+// parameter is specified (including specified as `None`), the decode function
+// returns the `default` value.
+// If `x` is not a valid JSON string and the optional, positional-only `default`
```

Comment:
Why is it positional only? Keyword-only would be clearer and more extensible.

If the answer is: for consistency with dict.pop, I'm not sure that's a compelling reason. That function is tiny and very familiar, any needs to avoid a dict allocation. None of that is true here.

Reply:
brandjon@ convinced me to do it positional-only in the Java Starlark version (which we already committed) for consistency with getattr, dict.get, and dict.pop. Arguably, consistency is as important as clarity.

As for extensibility, I was worried about that too until I realized that having an optional positional-only parameter does not preclude us from adding keyword-only parameters later if needed.

Reply:
The Java version could be quickly updated if we decide to change things here. I'll let you and Jon decide.

Reply:
Discussed with @tetromino offline. I'm pretty ambivalent. Consistency with `dict.get` and `getattr` is nice. On the other hand there may be other library APIs that need a default and for which a keyword is more clearly appropriate, so we may be sacrificing consistency with those. So I'll defer to @tetromino on this.

> having an optional positional-only parameter does not preclude us from adding keyword-only parameters later if needed

If you really mean keyword-*only*, we'd have to break code relying on the positional convention at some point.

Reply:
I would resolve the bike-shed color question by making the `default` param keyword or positional. (And will make the corresponding change in Java.)